app/components/settings_field/settings_field.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module SettingsField
+  class SettingsField < ApplicationComponent
+    def initialize(type:, setting:, **)
+      super
+
+      @type = type
+      @setting = setting
+    end
+
+    def call
+      component('field', object: Setting.new, value: Setting.send(setting), attr: setting) do |field|
```

Comment:
Why is the object always a new instance of the `Setting` class and not the `Setting` class itself?

And I was a bit confused as to what `Setting.send(setting)` does until I understood that `setting` refers to the name of the setting attribute that is referred to in this field. As `setting` could also describe the `Setting` which holds all `settings` / `Setting attributes` it might make it more clear to change the name of `setting` to `setting_name` or `setting_attribute`?

Reply:
> Why is the object always a new instance of the Setting class and not the Setting class itself?

This is a workaround that was already in place before ([see this line](https://github.com/tactilenews/100eyes/pull/1266/files#diff-bbc928fe5608c64d9a1d98217562d6b8a4fbbc5c8bc0d17335e3ae3825bdad2fL8)). I think this is due to the way the `Field` component derives the `id` and `name` attributes from the model name ([source](https://github.com/tactilenews/100eyes/blob/master/app/components/field/field.rb#L43)), as it uses ActiveRecord’s `model_name` method which is only available on ActiveRecord instances, not classes.

Reply:
> And I was a bit confused as to what Setting.send(setting) does until I understood that setting refers to the name of the setting attribute that is referred to in this field. As setting could also describe the Setting which holds all settings / Setting attributes it might make it more clear to change the name of setting to setting_name or setting_attribute?

I’d like to keep the component API simple and short, that’s why I went with `setting`:

```rb
<%= c('settings_field', type: :url_input, setting: :onboarding_logo) %>
<%= c('settings_field', type: :input, setting: :onboarding_byline) %>
```

How about `attr` as an alternative?

```rb
<%= c('settings_field', type: :url_input, attr: :onboarding_logo) %>
<%= c('settings_field', type: :input, attr: :onboarding_byline) %>
```

(We could also use `attribute` instead of `attr`, but we already use the abbreviated from in other places of the app.)

Reply:
> > Why is the object always a new instance of the Setting class and not the Setting class itself?
> 
> This is a workaround that was already in place before ([see this line](https://github.com/tactilenews/100eyes/pull/1266/files#diff-bbc928fe5608c64d9a1d98217562d6b8a4fbbc5c8bc0d17335e3ae3825bdad2fL8)). I think this is due to the way the `Field` component derives the `id` and `name` attributes from the model name ([source](https://github.com/tactilenews/100eyes/blob/master/app/components/field/field.rb#L43)), as it uses ActiveRecord’s `model_name` method which is only available on ActiveRecord instances, not classes.

Makes sense, thanks for the explanation!

Reply:
> How about attr as an alternative?

Seems like a good solution to me. It then also conforms nicely with the `field` API, where we would pass  `attr` to `attr`:

` component('field', object: Setting.new, value: Setting.send(attr), attr: attr)`

Much more readable to me. 👍